### PR TITLE
add project and group access token

### PR DESCRIPTION
**Pull request by beechesII**
added possibility to check also:
* Group Access Token
* Project Access Token

### CODE REVIEW

File: check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py [hunk 1]
```diff
@@ -31,7 +33,15 @@
...
 
-access_tokens = gl.personal_access_tokens.list(user_id=args.user_id, lazy=True, active=True)
+if args.user_id != None:
+    access_tokens = gl.personal_access_tokens.list(user_id=args.user_id, lazy=True, active=True)
+elif args.project_id != None:
+    access_tokens = gl.projects.get(args.project_id, lazy=True, active=True).access_tokens.list()
+elif args.group_id != None:
+    access_tokens = gl.groups.get(args.group_id, lazy=True, active=True).access_tokens.list()
+else:
+    print('Please define on of these arguments: --user_id, --project_id, --group_id')
```
Comment (schurzi) [inline check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py:43]:
```suggestion
    print('Please define one of these arguments: --user_id, --project_id, --group_id')
```

**Review by xFuture603 — APPROVED** [review]

File: check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py [hunk 2]
```diff
@@ -17,7 +17,9 @@
 
 parser.add_argument('--url', required=True)
 parser.add_argument('--private_token',required=True)
-parser.add_argument('--user_id',required=True)
+parser.add_argument('--user_id', default=None)
+parser.add_argument('--project_id', default=None)
+parser.add_argument('--group_id', default=None)
```
Comment (schurzi) [inline check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py:22]: WDYT using one argument `scope` and adding vlaues for each scope (https://docs.python.org/3/library/argparse.html#choices) then you would not need to do error handling.

Comment (beechesII) [reply check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py]: I changed the arguments from user_id, project_id and group_id to scope and id.

File: check-plugins/check_gitlab_token_expiration/README.md [hunk 3]
```diff
@@ -1,12 +1,22 @@
 # Check Gitlab Token Expiration
 
-This script is an icinga2 monitoring check to check when a Gitlab token will expire.
+This script is an icinga2 monitoring check to check when a personal access token, a project access token or a group access will expire.
```
Comment (schurzi) [inline check-plugins/check_gitlab_token_expiration/README.md:3]:
```suggestion
This script is an icinga2 monitoring check to check when a personal, project or group access token will expire.
```

missing `token`, maybe shorten a bit

File: check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 """
 Copyright 2023-2024 Deutsche Telekom MMS GmbH
 Maintainer: Christopher Grau
-check when a private token will expire
+check when a personal access token, a project access token or a group access token will expire
```
Comment (schurzi) [inline check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py:6]:
```suggestion
check when a personal, project or group access token will expire
```

File: check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py [hunk 5]
```diff
@@ -12,17 +12,24 @@
 import gitlab
 
 parser = argparse.ArgumentParser(
-  prog = "check_gitlab_token_expiration.py"
-  )
+  prog = "check_gitlab_token_expiration.py",
+  description = "check when a personal access token,a project access token or a group access token will expire"
```
Comment (schurzi) [inline check-plugins/check_gitlab_token_expiration/check_gitlab_token_expiration.py:16]:
```suggestion
  description = "check when a personal, project or group access token will expire"
```

**Review by schurzi — DISMISSED** [review]
slight wording issue, everything else is ok

**Review by schurzi — APPROVED** [review]

**Pull request merged**